### ISSUE TITLE
Is there updated documentation for setting up teams apps?

### ISSUE THREAD

**Issue by xstos** [MicrosoftDocs/msteams-docs#2981]
I'm following the tutorial and the steps have diverged from the microsoft teams toolkit in vscode. Is there more current documentation anywhere?

Thanks,

Chris

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 8f876156-cc1e-468f-4647-5143a5de03b1
* Version Independent ID: 618f43a3-3371-ad99-13aa-087b9303298c
* Content: [Get started - Build and run your first app - Teams](https://docs.microsoft.com/en-us/microsoftteams/platform/build-your-first-app/build-and-run)
* Content Source: [msteams-platform/build-your-first-app/build-and-run.md](https://github.com/MicrosoftDocs/msteams-docs/blob/master/msteams-platform/build-your-first-app/build-and-run.md)
* Product: **msteams**
* GitHub Login: @heath-hamilton
* Microsoft Alias: **lajanuar**

**Comment by RamaMohanaChary** [comment]
Hi @xstos, 

Currently we don't have any other other document, could you please let us know the exact gap that you are facing in document.

**Comment by xstos** [comment]
I tried to create a single sign on channel app, but it's asking me for "Azure AD single sign-on" FQDN for my app. I am just testing however and it won't let me enter localhost. Do you have any guidance as to how to test these apps before deploying them formally? Do I have to serve this thing on our company domain in order to test? If I put a fake URL in there does it matter? Should localhost work? There are so many configuration options (especially in App Studio) for which I can never tell what matters and what doesn't for debugging purposes. Also, do I need to create separate scaffoldings for all the various app types, or can I just create a master addin that does all of them by following all the separate tutorials and then merging all their artifacts together? Some guidance would be appreciated for beginners like me that come from VSTO development and are overwhelmed by the myriads of permutations of integrations and options.

**Comment by RamaMohanaChary** [comment]
Hi @xstos,

Could you please try with the sample code for [BOT SSO](https://github.com/microsoft/BotBuilder-Samples/tree/main/experimental/teams-sso/csharp_dotnetcore) and follow the steps in ReadME file. It would help you.

**Comment by xstos** [comment]
I won't have time to get back to teams stuff for a while (since we're currently battling SPFX). Shall we close this ticket and I can reopen one later on? The size of node_modules in these addins is insane. We've spent a week fighting with yarn workspaces trying to import our existing Word/Outlook 365 plugin into SPFX. 200K files for a handful of addins seems excessive.

**Comment by Mamatha-MSFT** [comment]
@xstos As of now you can close the ticket from your end and when ever you feel it requires, then create a new ticket.

**Comment by xstos** [comment]
So I found a work around to have the tutorial documentation screenshots match up with the vscode teams addin. I just downgraded my vscode addin to v1.1.0 and this allowed me to blend various addins together like in the tutorial screenshot.

[https://docs.microsoft.com/en-us/microsoftteams/platform/assets/images/build-your-first-app/choose-tab.png](https://docs.microsoft.com/en-us/microsoftteams/platform/assets/images/build-your-first-app/choose-tab.png)

Not sure why this feature was removed. In fact, as I tested out various addin versions, the addin UX varied dramatically...
